src/MatchResult/AbstractMatchResult.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+
+namespace Plesk\Wappspector\MatchResult;
+
+use JsonSerializable;
+use League\Flysystem\PathTraversalDetected;
+use League\Flysystem\WhitespacePathNormalizer;
+
+abstract class AbstractMatchResult implements MatchResultInterface, JsonSerializable
+{
+    public function __construct(
+        protected string $path,
+        protected ?string $version = null,
+        protected ?string $application = null,
+    ) {
+        try {
+            $this->path = (new WhitespacePathNormalizer())->normalizePath($this->path);
+        } catch (PathTraversalDetected) {
+            $this->path = '/';
+        }
+    }
+
+
+    public function getPath(): string
+    {
+        return $this->path;
+    }
+
+    public function getVersion(): ?string
+    {
+        return $this->version;
+    }
+
+    public function getApplication(): ?string
+    {
+        return $this->application;
+    }
+
+    public function jsonSerialize(): array
```

Comment:
It looks like this name doesn't fit the aim of this function because of the return type (array).
JSON is `string`.

Reply:
its required to be this way for [\JsonSerializable](https://www.php.net/manual/en/jsonserializable.jsonserialize.php) interface